lesson4/test_api/test_filtering_resources.py

```diff
@@ -0,0 +1,61 @@
+# Позитивные проверки - Поиск по id, должен вернуться 1 ресурс
+def test_filtering_positive_by_id(session, base_url, fixture_id_resource_positive):
+    res = session.get(url=f'{base_url}?id={fixture_id_resource_positive}')
+    assert res.status_code == 200, "Status code is not 200"
+    assert len(res.json()) == 1, "Length is more than 1"
+    assert res.json()[0]["id"] == fixture_id_resource_positive, "Wrong id"
+
+
+# Позитивные проверки - Поиск по userId, проверка что в каждом словаре в массиве указанный userId
+def test_filtering_positive_by_userid(session, base_url, fixture_test_filtering_positive_by_userid):
+    res = session.get(url=f'{base_url}?userId={fixture_test_filtering_positive_by_userid}')
+    assert res.status_code == 200, "Status code is not 200"
+    r = res.json()
+    for i in range(len(r)):
+        assert r[i]["userId"] == fixture_test_filtering_positive_by_userid, "Wrong userId"
+
+
+# Позитивные проверки - Поиск по completed, проверка что в каждом словаре в массиве указанный completed
+def test_filtering_positive_by_completed(session, base_url, fixture_test_filtering_positive_by_completed):
+    res = session.get(url=f'{base_url}?completed={fixture_test_filtering_positive_by_completed}')
+    assert res.status_code == 200, "Status code is not 200"
+    r = res.json()
+    for i in range(len(r)):
+        assert r[i]["completed"] == fixture_test_filtering_positive_by_completed, "Wrong completed"
```

Comment:
Here you can use iteration without range, for instance:

    todos = res.json()
    for t in todos:
        assert t["completed"] == fixture_test_filtering_positive_by_completed, "Wrong completed"